controllers/networktopology.go

```diff
@@ -72,9 +75,10 @@ func (r *NetworkTopologyReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, nil
 	}
 
-	if !r.client.HasStatusCondition(ctx, cluster) {
+	if !capiconditions.Has(cluster, networkTopologyCondition) {
+		capiconditions.MarkFalse(cluster, capi.ConditionType(networkTopologyCondition), "InProgress", capi.ConditionSeverityInfo, "")
 		// We're ok to continue if this fails
-		_ = r.client.UpdateStatusCondition(ctx, cluster, corev1.ConditionFalse)
+		_ = r.client.UpdateStatus(ctx, cluster)
```

Comment:
I think this can be removed now, as you added deferred Update.

Reply:
Different function. This one we want to set early on, before the reconcile has done anything.